storage/innobase/handler/ha_innodb.cc

```diff
@@ -15212,6 +15212,15 @@ ha_innobase::check(
 		m_prebuilt->need_to_access_clustered =
 			!!(check_opt->flags & T_EXTEND);
 
+#ifdef WITH_INNODB_SCN
+		if (m_prebuilt->need_to_access_clustered && innodb_use_scn) {
+		  /* trx id on secondary index page can't be used to accurately determine
+		  visibility while scn is used, so disable it and print a warning */
+		  m_prebuilt->need_to_access_clustered = FALSE;
+		  sql_print_warning("InnoDB: check table ... extended is not supported by scn");
+		}
+#endif
```

Comment:
What is the exact problem, and would it also affect `lock_sec_rec_some_has_impl()` or `row_vers_impl_x_locked_low()`?

If I understood it correctly, `innodb_use_scn=true` causes the `DB_TRX_ID` in clustered index records to be something else than the last transaction identifier that modified the record. This could potentially break some logic in `trx_undo_prev_version_build()`.

I did not find any change to the way how `PAGE_MAX_TRX_ID` is updated on secondary index pages. The `CHECK TABLE…EXTENDED` code is simply traversing the entire available history of clustered index record versions to find out whether a secondary index record is orphan or belongs to some available record version. Why would that not work when `innodb_use_scn=true`?